x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvPSeriesWeightedSum.java

```diff
@@ -0,0 +1,164 @@
+
+        description = "Converts a multivalued expression into a single-valued column by multiplying every "
+            + "element on the input list by its corresponding term in P-Series and computing the sum.",
+        examples = @Example(file = "mv_pseries_weighted_sum", tag = "example")
+    )
+    public MvPSeriesWeightedSum(
+        Source source,
+        @Param(name = "number", type = { "double" }, description = "Multivalue expression.") Expression field,
+        @Param(
+            name = "p",
+            type = { "double" },
+            description = "It is a constant number that represents the 'p' parameter in the P-Series. "
+                + "It impacts every element's contribution to the weighted sum."
+        ) Expression p
+    ) {
+        super(source, Arrays.asList(field, p));
+        this.field = field;
+        this.p = p;
+    }
+
+    private MvPSeriesWeightedSum(StreamInput in) throws IOException {
+        this(Source.readFrom((PlanStreamInput) in), in.readNamedWriteable(Expression.class), in.readNamedWriteable(Expression.class));
+    }
+
+    @Override
+    protected TypeResolution resolveType() {
+        if (childrenResolved() == false) {
+            return new TypeResolution("Unresolved children");
+        }
+
+        TypeResolution resolution = isType(field, t -> t.isNumeric() && isRepresentable(t), sourceText(), FIRST, "numeric");
+
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+
+        resolution = TypeResolutions.isType(p, dt -> dt == DOUBLE, sourceText(), SECOND, "double");
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+
+        return resolution;
+    }
+
+    @Override
+    public boolean foldable() {
+        return field.foldable() && p.foldable();
+    }
+
+    @Override
+    public EvalOperator.ExpressionEvaluator.Factory toEvaluator(Function<Expression, ExpressionEvaluator.Factory> toEvaluator) {
+        return switch (PlannerUtils.toElementType(field.dataType())) {
+            case DOUBLE -> new MvPSeriesWeightedSumDoubleEvaluator.Factory(source(), toEvaluator.apply(field), toEvaluator.apply(p));
+            case NULL -> EvalOperator.CONSTANT_NULL_FACTORY;
+
+            default -> throw EsqlIllegalArgumentException.illegalDataType(field.dataType());
+        };
+    }
+
+    @Override
+    public Expression replaceChildren(List<Expression> newChildren) {
+        return new MvPSeriesWeightedSum(source(), newChildren.get(0), newChildren.get(1));
+    }
+
+    @Override
+    protected NodeInfo<? extends Expression> info() {
+        return NodeInfo.create(this, MvPSeriesWeightedSum::new, field, p);
+    }
+
+    @Override
+    public DataType dataType() {
+        return field.dataType();
+    }
+
+    @Evaluator(extraName = "Double", warnExceptions = { InvalidArgumentException.class })
+    static void process(DoubleBlock.Builder builder, int position, DoubleBlock block, double p) {
```

Comment:
If you make the `double p` a `@Fixed` then we can pass the `p` in as a real constant to the evaluator.